### PR TITLE
`StringSourceLocator` is forced to be built with a `non-empty-string`

### LINKED ISSUE #932: Enforce `StringSourceLocator` to be built with a `non-empty-string`
Currently, a `StringSourceLocator` can be constructed with an empty string, but will crash at runtime when doing so:

https://github.com/Roave/BetterReflection/blob/f3966ba125e3b4db948a240e8b29bc28882623c1/src/SourceLocator/Type/StringSourceLocator.php#L23-L38

We should really get rid of the exception, and instead enforce `non-empty-string` as input.

### PR DESCRIPTION
**Pull request by kukulich**
Fixes #932

I've started with 6.0.0 cleanup.

### CODE REVIEW

**Review by Ocramius — APPROVED** [review]
LGTM. @asgrim this removes a runtime exception, so it could be seen as a BC break, are you fine with it?

File: src/SourceLocator/Exception/EmptyPhpSourceCode.php [hunk 1]
```diff
@@ -1,11 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Roave\BetterReflection\SourceLocator\Exception;
-
-use InvalidArgumentException;
-
-class EmptyPhpSourceCode extends InvalidArgumentException
```
Comment (Ocramius) [inline src/SourceLocator/Exception/EmptyPhpSourceCode.php:9]: I think we deprecate and remove this in the next major

Comment (kukulich) [reply src/SourceLocator/Exception/EmptyPhpSourceCode.php:9]: So one PR for 5.7 with `@deprecated` and this one should target 6.0.0?

Comment (Ocramius) [reply src/SourceLocator/Exception/EmptyPhpSourceCode.php:9]: as you prefer on that: next major release branch is not yet open

Comment (kukulich) [reply src/SourceLocator/Exception/EmptyPhpSourceCode.php:9]: I would be happy to just open 6.0 branch so I can target PR for #932, #981 and #1073

Comment (Ocramius) [reply src/SourceLocator/Exception/EmptyPhpSourceCode.php:9]: Works for me: beware that automatic releases will stop creating 5.x branches, once the new major branch is here

Comment (kukulich) [reply src/SourceLocator/Exception/EmptyPhpSourceCode.php:9]: Is there a way how I can open the branch? Or will you please do it for me? :)

Comment (Ocramius) [reply src/SourceLocator/Exception/EmptyPhpSourceCode.php:9]: Oh, it already exists: https://github.com/Roave/BetterReflection/tree/6.0.x

Never mind 😁

**Review by Ocramius — APPROVED** [review]
Thanks @kukulich!

**Pull request merged**